lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/model/GraphContainer.java

```diff
@@ -62,17 +62,21 @@
      *
      * @param graph
      *            The initial graph
+     * @param reference
+     *            Reference currently active in the graph controller.
      */
-    public GraphContainer(final Graph<SequenceSegment> graph) {
+    public GraphContainer(final Graph<SequenceSegment> graph,
+            final Sequence reference) {
         this.graph = graph;
-
-        // TODO: Temporary line until sequence selection is implemented.
-        Sequence reference = this.graph.getSources().iterator().next()
-                .getSources().iterator().next();
-
+        for (SequenceSegment segment : this.graph.getAllVertices()) {
+            segment.setReferenceStart(1);
```

Comment:
This is incredibly inefficient. You're looping over all the vertices to set values to their initial values. At the very least move this into a different method and only call it when needed.